### PR TITLE
#18616 this preserves the inode when converting a relationship

### PR DESCRIPTION
**Pull request by wezell**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/business/RelationshipAPIImpl.java [hunk 1]
```diff
@@ -496,23 +508,35 @@ private void migrateOldRelationshipReferences(Relationship oldRelationship,
...
+     * @param isRequired
+     * @param childContentTypeVariable
+     * @return
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
     private com.dotcms.contenttype.model.field.Field createRelationshipField(final String fieldName, final String parentContentTypeID, final int cardinality, final boolean isRequired, final String childContentTypeVariable)
             throws DotDataException, DotSecurityException {
-        final com.dotcms.contenttype.model.field.Field field = FieldBuilder.builder(RelationshipField.class)
+        final com.dotcms.contenttype.model.field.RelationshipField field = ImmutableRelationshipField.builder()
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/RelationshipAPIImpl.java:531]: Codacy Issue found: [Unnecessary use of fully qualified name 'com.dotcms.contenttype.model.field.RelationshipField' due to existing import 'com.dotcms.contenttype.model.field.RelationshipField'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5653044)

File: dotCMS/src/main/java/com/dotmarketing/business/RelationshipAPIImpl.java [hunk 2]
```diff
@@ -451,14 +455,22 @@ public void convertRelationshipToRelationshipField(final Relationship oldRelatio
...
-        createRelationshipField(parentFieldName, childContentType.id(),
+        final com.dotcms.contenttype.model.field.Field childRelationshipField =createRelationshipField(parentFieldName, childContentType.id(),
                 oldRelationship.getCardinality(), oldRelationship.isParentRequired(),
                 parentContentType.variable() + "." + parentRelationshipField.variable());
 
-        //Get the new Relationship
-        final Relationship newRelationship = byTypeValue(
-                parentContentType.variable() + "." + parentRelationshipField.variable());
 
+        Relationship newRelationship = (Relationship) Try.of(()->  BeanUtils.cloneBean(oldRelationship)).getOrElseThrow(e-> new DotRuntimeException(e));
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/RelationshipAPIImpl.java:463]: Codacy Issue found: [Local variable 'newRelationship' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5653044)

**Comment by nollymar** [comment]
The change seems correct. However, it requires ITs and we need to make sure it doesn't break PP related content

**Review by freddyucv — CHANGES_REQUESTED** [review]
any test?

**Comment by wezell** [comment]
I am not sure this is the right solution.  Nolly will be taking a look at this.

**Review by dsilvam — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

**Pull request merged**